src/conduits/RwaInputConduit3.t.sol

```diff
@@ -0,0 +1,315 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+//
+// RwaUrn.t.sol -- Tests for the Urn contract
+//
+// Copyright (C) 2020-2021 Lev Livnev <lev@liv.nev.org.uk>
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "forge-std/Test.sol";
+import "ds-token/token.sol";
+import "ds-math/math.sol";
+import "ds-value/value.sol";
+
+import {Vat}              from "dss/vat.sol";
+import {Jug}              from "dss/jug.sol";
+import {Spotter}          from "dss/spot.sol";
+import {Vow}              from "dss/vow.sol";
+import {GemJoin, DaiJoin} from "dss/join.sol";
+import {Dai}              from "dss/dai.sol";
+
+import {RwaInputConduit3}  from "./RwaInputConduit3.sol";
+// import {RwaOutputConduit} from "../conduits/RwaOutputConduit3.sol";
+
+import {DssPsm}           from "dss-psm/psm.sol";
+import {AuthGemJoin5}     from "dss-psm/join-5-auth.sol";
+
+
+contract TestToken is DSToken {
+
+    constructor(string memory symbol_, uint8 decimals_) public DSToken(symbol_) {
+        decimals = decimals_;
+    }
+
+}
+
+contract TestVat is Vat {
+    function mint(address usr, uint256 rad) public {
+        dai[usr] += rad;
+    }
+}
+
+contract TestVow is Vow {
+    constructor(address vat, address flapper, address flopper)
+        public Vow(vat, flapper, flopper) {}
+    // Total deficit
+    function Awe() public view returns (uint256) {
+        return vat.sin(address(this));
+    }
+    // Total surplus
+    function Joy() public view returns (uint256) {
+        return vat.dai(address(this));
+    }
+    // Unqueued, pre-auction debt
+    function Woe() public view returns (uint256) {
+        return sub(sub(Awe(), Sin), Ash);
+    }
+}
+
+contract TestUrn {
+    function balance(address gem) public view returns (uint256) {
+        return DSToken(gem).balanceOf(address(this));
+    }
+}
+
+contract RwaInputConduit3Test is Test, DSMath {
+    address me;
+
+    TestVat vat;
+    Spotter spot;
+    TestVow vow;
+    DSValue pip;
+    TestToken usdx;
+    DaiJoin daiJoin;
+    Dai dai;
+
+    AuthGemJoin5 joinA;
+    DssPsm psmA;
+    RwaInputConduit3 inputConduit;
+    TestUrn testUrn;
+
+    bytes32 constant ilk = "usdx";
+
+    uint256 constant TOLL_ONE_PCT = 10 ** 16;
+    uint256 constant USDX_WAD = 10 ** 6;
```

Comment:
`WAD` literally means `10**18`. We should probably not be using this term here.
```suggestion
    uint256 constant USDX_BASE_UNIT = 10 ** 6;
```

Reply:
`wad`: some quantity of tokens, usually as a fixed point integer with 10^18 decimal places.

Reply:
This just an amount of some token

Reply:
Hmmm... yeah, that's the definition on the Glossary.
Maybe I'm being too nitpicky since this is just a test file, but when it comes to actual code, anything different that 10**18 is called something else (see the different join adapters for instance).

IDK, I'll leave it up to you.